### PR TITLE
ci: add os type to job name

### LINKED ISSUE #253: Duplicate PR steps are confusing
There are two PR steps for unit tests and two for E2E tests. The name should include the OS to distinguish them. Currently it looks confusing as to why there are duplicates. Also the PR check for windows seem to skip steps

Screenshot 2024-04-27 at 8 27 04 PM

This is not urgent, but nice to have.

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* Adding OS type to job name

### Related Issues

* Closes #253

### CODE REVIEW

**Review by rbarkerSL — DISMISSED** [review]
LGTM

**Comment by leninmehedy** [comment]
@JeffreyDallas Are those windows tests working? I can see most of the steps are skipped: https://github.com/hashgraph/solo/actions/runs/8988531067/job/24689636867?pr=257

**Review by jeromy-cannon — APPROVED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

File: .github/workflows/zxc-compile-code.yaml [hunk 1]
```diff
@@ -64,13 +64,13 @@ permissions:
...
   compile:
-    name: ${{ inputs.custom-job-label || 'Compiles' }}
-    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os:
           - windows-2022
           - [self-hosted, Linux, large, ephemeral]
+    runs-on: ${{ matrix.os }}
+    name: "(${{ join(matrix.os, ',') }})"
```
Comment (jeromy-cannon) [inline .github/workflows/zxc-compile-code.yaml:73]: Nathan gave me the feedback to have name first, then runs-on

Comment (JeffreyDallas) [reply .github/workflows/zxc-compile-code.yaml:72]: Updated

**Comment by JeffreyDallas** [comment]
The change already added by Jereomy, no longer need this PR

**Pull request closed without merging**